BLSQ/iaso
# Polio 1601 im pipeline

Refactor Im country view to use data from json data store

Related JIRA tickets :  POLIO-1601

## Self proofreading checklist


## Doc
N/A

## Changes
- Expand and rename  `LQASCountryViewset` to handle IM:
    - rename viewset, url and file to reference IM explicitly
    - enforce stricter condition on slug matching to be able to handle 3 types of IM results
    - update test and remove obsolete test
    - update LQAS API url in front-end code
- Refactor Im to follow the same structure as LQAS : overview > Map + list
    - Add `ImCountryListOverview`
    - Add `ImCountryOverview`
    - Add `ImCountryMap`
    - Move` ImSummary`
    - Delete `LqasImMap`
    - Add constant file to IM
- regroup chart related components
    - move lqas specific components out of shared folder
    - move useLqasIm in `hooks/api` folder
- Deduplicate code:
    - Add `useSelectedRound` hook to handle round selection and change
    - move `useGetCampaigns` call out of `useLqasData` and `useImData` to enable use of `useSelectedRounds`
    - Add single map component `LqasImCountryMap` and use it in `LQasOverviewContainer` and `ImOverviewContainer`
    - add `useStyle` hook in shared folder
    - Add `useMapShapes`, `useLqasImTabState` hooks
    - Add `LqasImTabs` component
- Fix some routing issues with in household IM
- Prepare code for new IM status:
    - add constant
    - add shape styling for error status (hashed pattern)
    - add status to status calculations
- Fix bug with render of round 0 in summary dropdown list
- Fix bug with `DatesIgnored` component
- Remove some useless react fragments
- Add constants for IM api urls
- improve some TS typing
- Format numbers in Summary

## How to test

- Choose a country
- Go go to polio staging. Copy the json data store entries for this country in the django admin and replace those in your DB with those from staging. Data store key formats are `im_hh_ohh_<orgunit_id>`, `im_ohh_<orgunit_id>`,`im_hh_<orgunit_id>`. Make sure to delete any duplicate data store key to avoid errors
- To make sure the data will display correctly, make sure the scopes on the local DB are the same as on staging. Same with the round/IM dates
- Once everything is setup, just go to polio> IM and test all 3 pages (Global, in household, out of household)

## Print screen / video

Out of household

https://github.com/BLSQ/iaso/assets/38907762/d6e763f8-4663-4edb-9729-6c8df559c734

In household

https://github.com/BLSQ/iaso/assets/38907762/a8fd58ca-6e07-4717-aa3c-3690d1907b25

Global

https://github.com/BLSQ/iaso/assets/38907762/dc5d9d36-97ba-43e9-a6be-4645ae96f46e


## Notes

The "Refresh" Button, that allows users to launch the pipeline and refresh data from the UI will be added as part of another PR
